### ISSUE TITLE
[icetray] error not descriptive (Trac #2093)

### ISSUE THREAD

**Issue by icecube-trac** [icecube-trac/tix2#2085]
When concatenating a .gz GCD file with a .bz2 i3 file together, steamshovel threw the following error, which is not very descriptive:

FATAL (icetray): Frame tag found is  (I3Frame.cxx:627 in bool I3Frame::load(IStreamT &, const vector<string> &, bool) [IStreamT = boost::iostreams::filtering_stream<boost::iostreams::input, char, std::__1::char_traits<char>, std::__1::allocator<char>, boost::iostreams::public_>])
libc++abi.dylib: terminating with uncaught exception of type std::runtime_error: Frame tag found is  (in bool I3Frame::load(IStreamT &, const vector<string> &, bool) [IStreamT = boost::iostreams::filtering_stream<boost::iostreams::input, char, std::__1::char_traits<char>, std::__1::allocator<char>, boost::iostreams::public_>])
Abort trap: 6

Migrated from https://code.icecube.wisc.edu/ticket/2093
```json
{
    "status": "closed", 
    "changetime": "2017-09-18T19:07:19", 
    "description": "When concatenating a .gz GCD file with a .bz2 i3 file together, steamshovel threw the following error, which is not very descriptive:\n\nFATAL (icetray): Frame tag found is  (I3Frame.cxx:627 in bool I3Frame::load(IStreamT &, const vector<string> &, bool) [IStreamT = boost::iostreams::filtering_stream<boost::iostreams::input, char, std::__1::char_traits<char>, std::__1::allocator<char>, boost::iostreams::public_>])\nlibc++abi.dylib: terminating with uncaught exception of type std::runtime_error: Frame tag found is  (in bool I3Frame::load(IStreamT &, const vector<string> &, bool) [IStreamT = boost::iostreams::filtering_stream<boost::iostreams::input, char, std::__1::char_traits<char>, std::__1::allocator<char>, boost::iostreams::public_>])\nAbort trap: 6\n", 
    "reporter": "dxu", 
    "cc": "", 
    "resolution": "wontfix", 
    "_ts": "1505761639503212", 
    "component": "combo core", 
    "summary": "[icetray] error not descriptive", 
    "priority": "minor", 
    "keywords": "", 
    "time": "2017-09-18T18:10:15", 
    "milestone": "", 
    "owner": "", 
    "type": "rumor / allegation"
}
```

**Comment by icecube-trac** [comment]
Trac update at 20170918T18:40:42: 
* olivas changed status from "new" to "closed"
* olivas changed resolution from "" to "wontfix"
* olivas commented:

> Unfortunately, I can't think of a good way to detect that the user has cat'ed two files together that were compressed with different algorithms.  You'd almost certainly get the same error if you had a perfectly healthy bz2 file with the incorrect extension '.gz' for example.  IceTray uses the extension to figure out how to decompress the file and it's on the user to make sure the extension is correct and consistent.  I'm fairly certain there are a lot of bad things that can happen to a file (especially when cat'ing them together) that could cause this error.
> 
> In short we could make that error less verbose, but not more descriptive.

**Comment by icecube-trac** [comment]
Trac update at 20170918T18:45:53: david.schultz commented:

> I think the problem is that "frame tag" means nothing to most users, because they don't know it exists.  Telling them "corrupt file or not an i3 file" is more meaningful.

**Comment by icecube-trac** [comment]
Trac update at 20170918T19:06:09: 
* david.schultz changed _comment0 from "Fixed in #2972." to "1505761639503212"
* olivas commented:

> Fixed in r2972/IceTray.
